### ISSUE TITLE
[Az.Synapse 0.13.0] Get-AzSynapseWorkspace resource provider 2021-04-01-preview unsupported in some clouds

### ISSUE THREAD

**Issue by cutecycle** [Azure/azure-powershell#15616]
We work in AzureUSGovernment and had a script that defaulted to the latest version of Az.Synapse and broke on a `Get-AzSynapseWorkspace` call. I think it's because of the update to the dotnet-sdk Synapse management package.

```
No registered resource provider found for location 'usgovvirginia' and API version '2021-04-01-preview' for type 'workspaces'. The supported api-versions are '2019-06-01-preview, 2020-12-01, 2021-03-01, 2021-05-01'.
```
## Steps to reproduce

```powershell
(Get-AzSynapseWorkspace -WorkspaceName "workspaceName").ConnectivityEndpoints.SqlOnDemand
```

## Environment data
(WIP)

https://github.com/Azure/azure-sdk-for-net/pull/21769#event-4957007194
https://github.com/Azure/azure-powershell/commit/0b8a9d22bbf5a81aca2677cfc3c22dc9e8e5c6ff
```

```

## Module versions
(WIP)

```powershell

```

## Debug output

```

```

## Error output

```

```

**Comment by idear1203** [comment]
Thanks for reaching out! I will do some investigation first.

**Comment by cutecycle** [comment]
Additionally: this cmdlet does not seem to throw the correct exit code when failing with this error.


```
Update-AzSynapseWorkspace : Operation returned an invalid status code 'NotFound' The resource type 'workspaces' could 
not be found in the namespace 'Microsoft.Synapse' for api version '2021-04-01-preview'. The supported api-versions are 
'2019-06-01-preview,2020-12-01,2021-03-01,2021-05-01,2021-06-01'.
At C:\agent\_work\_temp\c12e2fb1-3a83-47e2-bfff-34a598ffadc8.ps1:38 char:1
+ Update-AzSynapseWorkspace -WorkspaceName $workspaceName -SqlAdministr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Update-AzSynapseWorkspace], AzPSResourceNotFoundCloudException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Synapse.UpdateAzureSynapseWorkspace
 
##[debug]Exit code: 0
##[debug]Leaving Invoke-VstsTool.

```

**Comment by idear1203** [comment]
@dingmeng-xue could you please take a look at the exit code issue? Code link is here: [Update-AzSynapseWorkspace](https://github.com/Azure/azure-powershell/blob/main/src/Synapse/Synapse/Commands/ManagementCommands/Workspace/UpdateAzureSynapseWorkspace.cs) . We don't have control about the exit code?

**Comment by dingmeng-xue** [comment]
The error message reflects message from service. You can see `The resource type 'workspaces' could 
not be found in the namespace 'Microsoft.Synapse' for api version '2021-04-01-preview'. The supported api-versions are 
'2019-06-01-preview,2020-12-01,2021-03-01,2021-05-01,2021-06-01'.` 

It seems `2021-04-01-preview` is not available in that Azure environment.

**Comment by idear1203** [comment]
@dingmeng-xue this is true and we are actively working on fix for the API version issue. However, I want to understand that when the exception is thrown, why the exit code of the PowerShell cmdlet is zero?

**Comment by dingmeng-xue** [comment]
Status Code is 404 which is `NotFound` literally.

**Comment by idear1203** [comment]
@cutecycle may I know how you get the following message:

```
##[debug]Exit code: 0
##[debug]Leaving Invoke-VstsTool.
```

These are not the output of the PowerShell cmdlet. If you want to check status code of the PowerShell cmdlet, please use `$?`. If the value is `False`, it indicates the last command isn't successful.

**Comment by cutecycle** [comment]
@idear1203 I think you might be right; this cmdlet is being wrapped in an Azure DevOps task which may not be retrieving the exit code properly
